### ISSUE TITLE
ausplotsR depends on retiring spatial infrastructure packages

### ISSUE THREAD

**Issue by rsbivand** [ternaustralia/ausplotsR#65]
You will be aware, for example from:
https://r-spatial.org/r/2022/04/12/evolution.html,
https://r-spatial.org/r/2022/12/14/evolution2.html,
https://r-spatial.org/r/2023/04/10/evolution3.html and
https://rsbivand.github.io/csds_jan23/bivand_csds_ssg_230117.pdf and
perhaps view
https://www.youtube.com/watch?v=TlpjIqTPMCA&list=PLzREt6r1NenmWEidssmLm-VO_YmAh$
that `rgdal`, `rgeos` and `maptools` will be retired this
year, in October 2023.

ausplotsR uses maptools::map2SpatialPolygons, but only uses it to generate
a SpatialPolygons object (aus.pl) twice, once in R/map_attribute.R, once
in map_ausplots.R. You could avoid relying on maptools (which will be gone
in just 6 months), by saving aus.pl as a serialised R object (RDS or RDA
file), and then drop map, mapdata and maptools (untried). Alternatively,
use the approach linked to in yesterday's blog in
https://github.com/r-spatial/evolution/blob/main/pkgapi_230305_refs.csv,
of coercing from map to sf, and on again to Spatial then to
SpatialPolygons, but this would entail importing from sf.
  
Note also that all map geometries have "+proj=longlat +ellps=clrk66
+no_defs +type=crs" as their CRS definition, they are definitely not
"+init=epsg:4326" and do not have a datum definition (except in North
America, where NAD27 is probably applicable). This conclusion is based on
the ellipsoid parameters given in C code in mapproj/src/map.h, ECC defined
as .08227185422, which is the same as:
```
ellps <- sf::sf_proj_info("ellps")
(clrk66 <- unlist(ellps[ellps$name=="clrk66",]))
eval(parse(text=clrk66["major"]))
eval(parse(text=clrk66["ell"]))
print(sqrt((a^2-b^2)/a^2), digits=10)
```

Also sent as email, since this repo is not exposed in DESCRIPTION.

**Comment by GregGuerin** [comment]
Roger,

thanks for taking the time to post this.

We will follow up to implement a solution such as one of those you mention.

**Comment by GregGuerin** [comment]
Solution:  maps, mapdata and maptools are no longer imported from at all and have been removed from namespace.

Instead, base maps are generated directly in ggplot using ggplot::map_data (same base data) and plotted in ggplot using the newer coord_sf argument to make spatial polygons.

258a5f1

20f2770

3fd3b54

**Comment by rsbivand** [comment]
I see `1.2.9` on CRAN released last month; is `2.0.0` coming soon?

**Comment by GregGuerin** [comment]
> I see `1.2.9` on CRAN released last month; is `2.0.0` coming soon?

Hello - yes, expecting to release `2.0.0` to CRAN by the end of this month (June), so we should be up to date with this issue.

Thanks
